stdlib/ballerina-mime/src/main/java/org/ballerinalang/mime/util/MimeUtil.java

```diff
@@ -292,12 +288,11 @@ public static String getTextPayload(BStruct entity) {
         if (isNotNullAndEmpty(returnValue)) {
             return returnValue;
         } else {
-            BStruct fileHandler = (BStruct) entity.getRefField(OVERFLOW_DATA_INDEX);
-            String filePath = fileHandler.getStringField(FILE_PATH_INDEX);
+            String filePath = getFilePathFromFileHandler(entity);
             try {
-                return new String(readFromFile(filePath), UTF_8);
+                return filePath.isEmpty() ? null : new String(readFromFile(filePath), UTF_8);
             } catch (UnsupportedEncodingException e) {
-                LOG.error("Error occured while extracting text payload from entity", e.getMessage());
+                LOG.error("Error occurred while extracting text payload from entity", e.getMessage());
```

Comment:
Shall we make this variable log? 
private static final Logger log = LoggerFactory.getLogger(MimeUtil.class);

Reply:
Addressed with b98823c.